tools/go-cli/go-whisk/whisk/client.go

```diff
@@ -276,6 +248,65 @@ func (c *Client) Do(req *http.Request, v interface{}) (*http.Response, error) {
     return resp, werr
 }
 
+func parseErrorResponse(resp *http.Response, data []byte, v interface{}) (*http.Response, error) {
+    Debug(DbgInfo, "HTTP failure %d + body\n", resp.StatusCode)
+    errorResponse := &ErrorResponse{Response: resp}
+    err := json.Unmarshal(data, errorResponse)
+
+    // If the body matches the error response format, return an error containing the response error information (#5).
+    // Otherwise, the body contents are unknown (#6).
+    if err == nil {
+        if errorResponse.Code == nil || errorResponse.ErrMsg == nil {
+            Debug(DbgInfo, "Error code, or error message is null\n")
+            return parseWhiskErrorResponse(resp, data, v)
+        } else {
+            Debug(DbgInfo, "HTTP failure %d; server error %s\n", resp.StatusCode, errorResponse)
+            werr := MakeWskError(errorResponse, resp.StatusCode - 256, DISPLAY_MSG, NO_DISPLAY_USAGE)
+            return resp, werr
+        }
+    } else {
+        Debug(DbgError, "HTTP response with unexpected body failed due to contents parsing error: '%v'\n", err)
+        var errStr = fmt.Sprintf("Request failed (status code = %d). Error details: %s", resp.StatusCode, data)
+        werr := MakeWskError(errors.New(errStr), resp.StatusCode - 256, DISPLAY_MSG, NO_DISPLAY_USAGE)
+        return resp, werr
+    }
+}
+
+func parseWhiskErrorResponse(resp *http.Response, data []byte, v interface{}) (*http.Response, error) {
+    whiskErrorResponse := &WhiskErrorResponse{}
+    err := json.Unmarshal(data, whiskErrorResponse)
+
+    // Determine if a whisk.error() response was received. Otherwise, the body contents are unknown (#6)
+    if err == nil && whiskErrorResponse.Response.Result.Error != nil {
+        Debug(DbgInfo, "Detected that a whisk.error(\"%s\") was returned\n", *whiskErrorResponse.Response.Result.Error)
+        errMsg := fmt.Sprintf("Received a whisk.error() (status code = %d). Error details: %s", resp.StatusCode, data)
```

Comment:
I think what needs to be displayed here instead of the raw `data` is the parsed error string from the response data json:
```
"response": {
  "result": {
    "error": "This error thrown on purpose by the action."
  },
  "success": false,
  "status": "application error"
},
```